javalib/src/main/scala/java/util/Collections.scala

```diff
@@ -534,22 +534,41 @@ object Collections {
 
   def list[T](e: Enumeration[T]): ArrayList[T] = {
     val arrayList = new ArrayList[T]
-    e.asScala.foreach(arrayList.add(_))
+    e.scalaOps.foreach(arrayList.add(_))
     arrayList
   }
 
   def frequency(c: Collection[_], o: AnyRef): Int =
-    c.asScala.count(_ === o)
+    c.scalaOps.count(_ === o)
 
   def disjoint(c1: Collection[_], c2: Collection[_]): Boolean = {
     if (c1.size < c2.size)
-      !c1.asScala.exists(elem => c2.contains(elem))
+      !c1.scalaOps.exists(elem => c2.contains(elem))
     else
-      !c2.asScala.exists(elem => c1.contains(elem))
+      !c2.scalaOps.exists(elem => c1.contains(elem))
   }
 
-  def addAll[T](c: Collection[_ >: T], elements: Array[AnyRef]): Boolean =
-    c.addAll(elements.asInstanceOf[Array[T]].toSeq.asJava)
+  def addAll[T](c: Collection[_ >: T], elements: Array[AnyRef]): Boolean = {
+    val elementsColl = new AbstractCollection[T] {
```

Comment:
Huh, couldn't you use standard array ops to make an Scala iterator out of this and then wrap it as a java iterator?